test/StreamJsonRpc.Tests/MarshalableProxyTests.cs

```diff
@@ -499,6 +523,40 @@ public async Task DisposeOnDisconnect()
         await disposed.WaitAsync(this.TimeoutToken);
     }
 
+    [Fact]
+    public async Task SubTypeMarshalable_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableWithSubTypes();
+        IMarshalableWithSubTypes? proxy = await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
+    }
+
+    [Fact]
+    public async Task SubTypeMarshalable_SubType1_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableSubType1();
+        IMarshalableSubType1? proxy = (IMarshalableSubType1?)await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
+        Assert.Equal(2, await proxy!.GetPlusOneAsync(1));
+    }
+
+    [Fact]
+    public async Task SubTypeMarshalable_SubType2_CanCallMethods()
+    {
+        this.server.ReturnedMarshalableWithSubTypes = new MarshalableSubType2();
+        IMarshalableSubType2? proxy = (IMarshalableSubType2?)await this.client.GetMarshalableWithSubTypesAsync();
+        Assert.Equal(1, await proxy!.GetAsync(1));
+        Assert.Equal(3, await proxy!.GetPlusTwoAsync(1));
```

Comment:
nit: no need to use the `!` null forgiveness operator on the second line. The compiler recognizes that the line above it has already sealed your fate (it'll throw, or prove that it's non-null.)